internal/controller/management/projects/projects.go

```diff
@@ -341,6 +357,64 @@ func (r *reconciler) ensureProjectAdminPermissions(
 	return nil
 }
 
+func (r *reconciler) ensureDefaultProjectRoles(
+	ctx context.Context,
+	project *kargoapi.Project,
+) error {
+	const viewerRoleName = "viewer"
+
+	logger := logging.LoggerFromContext(ctx).WithFields(log.Fields{
+		"project":   project.Name,
+		"name":      project.Name,
+		"namespace": project.Name,
+		"role":      viewerRoleName,
+	})
+
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      viewerRoleName,
+			Namespace: project.Name,
+			Annotations: map[string]string{
+				rbacapi.AnnotationKeyManaged: rbacapi.AnnotationValueTrue,
+			},
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"events", "serviceaccounts"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{kargoapi.GroupVersion.Group},
+				Resources: []string{"freights", "promotions", "stages", "warehouses"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{rolloutsapi.GroupVersion.Group},
+				Resources: []string{"analysisruns", "analysistemplates"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
```

Comment:
How about secrets? I think the secrets tab may not function.

Reply:
Oh... so I was thinking reading secrets is probably _beyond_ what an average viewer should be able to do.

Reply:
Yes, let's not add this to this PR. I'm thinking we can do something special in Kargo API about returning redacted / non-sensitive secrets to anyone with the viewer role in the project.